pkg/inframetadata/internal/hostmap/hostmap.go

```diff
@@ -183,6 +187,48 @@ func (m *HostMap) Update(host string, res pcommon.Resource) (changed bool, md pa
 	return
 }
 
+func (m *HostMap) UpdateFromMetric(host string, metric pmetric.Metric) {
+	var point pmetric.NumberDataPoint
+
+	// Take last available point
+	switch metric.Type() {
+	case pmetric.MetricTypeGauge:
+		lastIndex := metric.Gauge().DataPoints().Len() - 1
+		point = metric.Gauge().DataPoints().At(lastIndex)
```

Comment:
Is metric guaranteed to have at least 1 data point?

Reply:
This does not seem to be a requirement on either the spec or opentelemetry-proto, so I added a check for it.